eng/Versions.props

```diff
@@ -71,7 +71,7 @@
     <NpgsqlEntityFrameworkCorePostgreSQLPackageVersion>8.0.8</NpgsqlEntityFrameworkCorePostgreSQLPackageVersion>
     <!-- for templates -->
     <MicrosoftExtensionsHttpResiliencePackageVersionForNet8>$(MicrosoftExtensionsHttpResiliencePackageVersion)</MicrosoftExtensionsHttpResiliencePackageVersionForNet8>
-    <MicrosoftExtensionsHttpResiliencePackageVersionForNet9>$(MicrosoftExtensionsHttpResiliencePackageVersion)</MicrosoftExtensionsHttpResiliencePackageVersionForNet9>
+    <MicrosoftExtensionsHttpResiliencePackageVersionForNet9>9.0.0-preview.9.24507.7</MicrosoftExtensionsHttpResiliencePackageVersionForNet9>
```

Comment:
My understanding is that `Microsoft.Extensions.Http.Resilience` is on a "tip support" policy. Meaning that only the latest version is supported. (I think there maybe a nuance here where the `8.0` version is LTS, but I can't seem to find the public policy.)

Given the above, I think we should always be referencing the current/latest version of `Microsoft.Extensions.Http.Resilience` from our templates regardless of TFM chosen. Even if you are targeting `net8.0`, we would bring in the 9.0 version of `Microsoft.Extensions.Http.Resilience`.

See also: [Use 8.0 era dependencies for non net9.0 TFMs (dotnet/extensions#5470)](https://github.com/dotnet/extensions/pull/5470)

cc @joperezr

Reply:
Shouldn't our repo be using the 9.x version then too? Currently it's using the 8.x version.

Reply:
Yes. That's part of this as well.

Reply:
Should I hold off on a change here until we get consensus?

Reply:
I think there is an agreement here. As @eerhardt points out, we want to use the tip of this package both in net8 and net9. There is still the PR that he pointed out above which hasn't been merged but will be merged before 9.0 ships.